### ISSUE TITLE
[Bug]: Nesting two `and`s inside an `or` filter causes only the first `and` to be evaluated

### ISSUE THREAD

**Issue by jollygreenegiant** [supabase-community/supabase-kt#434]
### General Info

- [X] I checked for similar bug report
- [X] I am using the latest version
- [X] I checked the [troubleshooting](https://github.com/supabase-community/supabase-kt/wiki/Troubleshooting) page for similar problems
- [X] I enabled [logging](https://github.com/supabase-community/supabase-kt/wiki/Logging) and checked the logs

### Version(s)

2.0.3

### Target(s)

Common

### What happened? (include your code)

I have the following query:

```
val friendship = supabaseClient.from("friendships")
    .select {
        filter {
            or {
                and {
                    eq("user1", currentUser)
                    eq("user2", userId)
                }
                and {
                    eq("user1", userId)
                    eq("user2", currentUser)
                }
            }
        }
    }
```
which is evaluating to the following URL:

`https://yrqjisjprvaehgsdqxfr.supabase.co/rest/v1/friendships?or=(user1.eq.<currentUserId>,user2.eq.<userId>)&select=*`

The second `and` statement is being left out of the filter entirely, meaning only the first one is getting evaluated.  I'm not sure if this is a bug or if I'm misunderstanding the way these filters are supposed to be used, but it seems like this should work.

Happy to provide more info if needed.

**Comment by jan-tennert** [comment]
This is a bug, using a logical operator within another logical operator twice should work. I possibly fixed this, can you try out version `2.1.0-dev-1` (publishing), to see if it works?

**Comment by jollygreenegiant** [comment]
It seems to be working, thank you!
